R/utilities-data-validation.R

```diff
@@ -151,6 +151,45 @@ checkEpsilon <- function(epsilon) {
 }
 
 
+#' Check delta parameter
+#' 
+#' This method is to send the user a warning message if they entered a delta value
+#' that will not be used.
+#' 
+#' If the mechanism is NOT the stabiltiy mechanism and the user entered a delta value, 
+#' send a warning message saying the delta value with not be used and set the delta 
+#' value to NULL. If the stability mechanism is being used and the user entered a 
+#' delta value, set it as the delta value (the value of delta will be checked in the 
+#' stability mechanism). If the stability mechanism is being used and the user did 
+#' not enter a delta value, set the delta to the default value (2^-30).
+#' 
+#' @param mechanism The mechanism chosen by determineMechanism
+#' @param delta The delta value entered by the user, may be NULL
+#' 
+#' @return The value that will be used as the delta value for the statistic, may be NULL.
+
+checkDelta <- function(mechanism, delta=NULL) {
+    # If a mechanism that uses a delta value is NOT being used, set delta to 0.
+    # If a mechanism that uses a delta value is being used, return the entered value.
+    # If the user did not enter a value, return the default value.
+    # Throw an error if the user entered a value that will not be used.
+    if (!(mechanism %in% c('mechanismStability', 'mechanismGaussian', 'mechanismExponential'))) {
```

Comment:
exponential mech has delta=0.

Reply:
might want to think about how we'd incorporate advanced composition into this framework. Probably fine for now to not worry about that I think?

Reply:
Megan pointed out current gap function in exp mech does require delta... maybe keep I guess?

Reply:
actually decided to remove for now and deal with once we refactor exp mech